### PR TITLE
Read inputs

### PR DESCRIPTION
**Pull request by RutvikM3**
# Description

Addresses issue 39

Added JSON input reading, and test file, to replace the current Aether.in file

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change that adds functionality)

# How Has This Been Tested?

Tested results on sample file, included here

# Checklist:

- [N/A] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [N/A] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by aburrell** [comment]
Hi @RutvikM3, just an FYI that you can turn the boxes into checked boxes by putting an `x` between the square brackets.  Alternatively, you can leave them empty and then tick them off after the list is created:

- [x] Already ticked with an `x`
- [ ] Empty for you to tick

File: inputs/aether_in.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{"Debug": [1, "iVerbose"], 
+
+ "Planet":"Earth",
```
Comment (aburrell) [inline inputs/aether_in.json:3]: Should this line have a space after the colon?  All the other lines do.

File: inputs/aether_in.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ "Endtime": [2011, 6, 21, 12, 10, 0],
+
+ "F107File": "UA/inputs.f107.txt",
+
+ "BField": "dipole",
+
+ "Output": {
+    "States": 300,
+    "BField":0}
+}
```
Comment (aburrell) [inline inputs/aether_in.json:16]: Recommend putting a newline here to simplify github rendering.

Comment (aburrell) [reply inputs/aether_in.json:16]: Here and elsewhere.

File: src/file_readinput.cpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
+
+// (c) 2020, the Aether Development Team (see doc/dev_team.md for members)
+// Full license can be found in License.md
+
+/*
+  This is an example code to demonstrate how to use the json library.
+
```
Comment (aburrell) [inline src/file_readinput.cpp:7]: Should example code be placed in a directory that indicates this, or have `example` in the name, or something else that makes it easy to identify?  Or is this provided as a simple base routine that is meant to be improved and incorporated into the main code?  If the second, perhaps this comment should be made clearer.

**Review by aburrell — COMMENTED** [review]
Some suggestions from a person who rarely uses json.  Has it been run through a linter?

File: inputs/aether_in.json (hunk 1 above)
Comment (RutvikM3) [reply inputs/aether_in.json:3]: The space after the colon is just optional. I ran this test file through a JSON linter, and it passed the tests.

File: src/file_readinput.cpp (hunk 3 above)
Comment (RutvikM3) [reply src/file_readinput.cpp:7]: Thanks for pointing this out, it is the latter, where this code will be improved and incorporated soon. Will clear that up.

File: inputs/aether_in.json (hunk 1 above)
Comment (aburrell) [reply inputs/aether_in.json:3]: Great, thanks for checking!  In that case I recommend keeping a consistent style throughout the code.  So, pick either having a space or not having a space.

Comment (aaronjridley) [reply inputs/aether_in.json:3]: I recommend having spaces, just to be consistent with the rest of the code and to ease readability. It is a pain to go through and make sure that they are all consistent, but it ultimately helps.

**Comment by aaronjridley** [comment]
Rutvik - I am fine if you merge this into the develop branch.  We need to then figure out how we should use this internally in the code - whether we should change the Inputs class or what.

**Comment by RutvikM3** [comment]
> Rutvik - I am fine if you merge this into the develop branch. We need to then figure out how we should use this internally in the code - whether we should change the Inputs class or what.

When trying to merge, it tells me "only those with write access to this repository can merge pull requests". I guess to solve this either you or someone with access has to merge, or you can grant me access and I should be able to do it.

**Pull request merged**